src/ImageSharp.Formats.Tiff/TiffDecoderCore.cs

```diff
@@ -0,0 +1,452 @@
+                {
+                    for (int i = 0 ; i < result.Length ; i++)
+                        result[i] = (int)ToInt16(bytes, i * TiffConstants.SizeOfShort);
+                    break;
+                }
+                case TiffType.SLong:
+                {
+                    for (int i = 0 ; i < result.Length ; i++)
+                        result[i] = ToInt32(bytes, i * TiffConstants.SizeOfLong);
+                    break;
+                }
+                default:
+                    throw new ImageFormatException($"A value of type '{entry.Type}' cannot be converted to a signed integer.");
+            }
+
+            return result;
+        }
+
+        public string ReadString(ref TiffIfdEntry entry)
+        {
+            if (entry.Type != TiffType.Ascii)
+                throw new ImageFormatException($"A value of type '{entry.Type}' cannot be converted to a string.");
+
+            byte[] bytes = ReadBytes(ref entry);
+            
+            if (bytes[entry.Count - 1] != 0)
+                throw new ImageFormatException("The retrieved string is not null terminated.");
+
+            return Encoding.UTF8.GetString(bytes, 0, (int)entry.Count - 1);
+        }
+
+        public Rational ReadUnsignedRational(ref TiffIfdEntry entry)
+        {
+            if (entry.Count != 1)
+                throw new ImageFormatException($"Cannot read a single value from an array of multiple items.");
+
+            return ReadUnsignedRationalArray(ref entry)[0];
+        }
+
+        public SignedRational ReadSignedRational(ref TiffIfdEntry entry)
+        {
+            if (entry.Count != 1)
+                throw new ImageFormatException($"Cannot read a single value from an array of multiple items.");
+
+            return ReadSignedRationalArray(ref entry)[0];
+        }
+
+        public Rational[] ReadUnsignedRationalArray(ref TiffIfdEntry entry)
+        {
+            if (entry.Type != TiffType.Rational)
+                throw new ImageFormatException($"A value of type '{entry.Type}' cannot be converted to a Rational.");
+
+            byte[] bytes = ReadBytes(ref entry);
+            Rational[] result = new Rational[entry.Count];
+
+            for (int i = 0 ; i < result.Length ; i++)
+            {
+                uint numerator = ToUInt32(bytes, i * TiffConstants.SizeOfRational);
+                uint denominator = ToUInt32(bytes, i * TiffConstants.SizeOfRational + TiffConstants.SizeOfLong);
+                result[i] = new Rational(numerator, denominator);
+            }
+
+            return result;
+        }
+
+        public SignedRational[] ReadSignedRationalArray(ref TiffIfdEntry entry)
+        {
+            if (entry.Type != TiffType.SRational)
+                throw new ImageFormatException($"A value of type '{entry.Type}' cannot be converted to a SignedRational.");
+
+            byte[] bytes = ReadBytes(ref entry);
+            SignedRational[] result = new SignedRational[entry.Count];
```

Comment:
Isn't it possible to do buffered read without allocating new arrays? 
You can store them as pre-allocated members, if the arrays are expected to have a known upper limit. If the limit is not known and/or or the arrays are too big, you can reuse buffers with the following utility class:
```C#
class AutoGrowBuffer<T> : IDisposable
{
    private PinnedBuffer<T> currentBuffer;
    public AutoGrowBuffer(int initialSize)
    {
        this.currentBuffer = new PinnedBuffer<T>(initialSize);
    }

    // BufferSpan<T> should be very similar to BufferPointer<T>, but has a count value. I can implement it, if it fits your needs :)  
    public BufferSpan<T> GetBufferSpan(int length)
    {
        if (this.currentBuffer.Array.Length < length)
        {
            this.currentBuffer.Dispose();
            this.currentBuffer = new PinnedBuffer<T>(length);
        }
        return new BufferSpan<T>(this.currentBuffer, length);
    } 
  
   public void Dispose()
   {
       this.currentBuffer .Dispose();
   }
}

// USAGE:
private AutoGrowBuffer<SignedRational> signedRationalBuffer;

// BufferSpan<T> should be very similar to BufferPointer<T>, but has a count value. I can implement it, if it fits your needs :)
public BufferSpan<SignedRational> ReadSignedRationalSpan(ref TiffIfdEntry entry)
{
    ....
    BufferSpan<SignedRational> result = this.signedRationalBuffer.GetBufferSpan(entry.Count);
    ....
}
```